src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10195,9 +10195,19 @@ private bool IsQueryExpressionAfterFrom(bool mayBeVariableDeclaration, bool mayB
 
         private QueryExpressionSyntax ParseQueryExpression()
         {
+            return ParseQueryExpression(0);
+        }
+
+        private QueryExpressionSyntax ParseQueryExpression(uint precedence)
+        {
             this.EnterQuery();
             var fc = this.ParseFromClause();
             fc = CheckFeatureAvailability(fc, MessageID.IDS_FeatureQueryExpression);
+            if (precedence > 1 && IsStrict)
```

Comment:
See the enum, above, which more clearly documents the meaning of the values. `precedence > 0` would mean anything except a top-level expression, which is not the intent here.